aqt/helper.py

```diff
@@ -297,6 +297,30 @@ def to_version(qt_ver: str) -> Version:
     )
 
 
+def to_version_permissive(version_str: str) -> Version:
+    """Converts a version string with dots (5.X.Y, etc) into a semantic version.
+    If the version omits either the patch or minor versions, they will be filled in with zeros,
+    and the remaining version string becomes part of the prerelease component.
+    If the version cannot be converted to a Version, a ValueError is raised.
+
+    This function is intended to be used on Version tags in an Updates.xml file.
+
+    to_version_permissive('1.33.1-202102101246') => Version('1.33.1-202102101246')
+    to_version_permissive('1.33-202102101246') => Version('1.33.0-202102101246')    # tools_conan
+    to_version_permissive('2020-05-19-1') => Version('2020.0.0-05-19-1')            # tools_vcredist
```

Comment:
This particular behavior might be a mistake. This is meant to handle version strings that are actually dates, as in qt.tools.windows_kits_debuggers and qt.tools.vcredist, as mentioned in my comment in #282. It might be better to use Date objects in this case, and then rely on other code to handle exceptions in the case that a Date is being compared with a Version. This will be an issue in the case of qt.tools.windows_kits_debuggers (which is in tools_qtcreator); most of the tools in that directory have Versions, but the debugger kit has a Date. IMHO in this case, and the tools_vcredist case, the user really should be using the 'tool variant name' parameter to install the tool, since the 'tool variant name' is stable and not incremented with each new version.

Maybe the `--spec` parameter should only be allowed for tools like `tools_ifw` that have 'tool variant name's that change constantly?